core/database/interface.go

```diff
@@ -13,6 +13,21 @@ type DBGetter interface {
 	GetDB(namespace string) (TrackedDB, error)
 }
 
+// DBManager describes the ability to create and delete databases.
+type DBManager interface {
```

Comment:
why is this file called `interface.go`? seems too generic and we have `trackeddb.go` in the same package, so maybe this should be `dbgetter.go`?